src/api/l_math_vectors.c

```diff
@@ -112,6 +121,9 @@ int luax_readscale(lua_State* L, int index, vec3 v, int components, const char*
         }
       }
       return index;
+    case LUA_TTABLE:
+      luax_readobjarr(L, index, 3, v, "scale");
```

Comment:
Should be `luax_readobjarr(L, index, components, v, "scale");` so it reads the correct number of components (and maybe skip the table branch if components == 1?)